en/lessons/specifics/ecto/02_associations.md

```diff
@@ -0,0 +1,672 @@
+
+%Moviedb.Movie{
+  __meta__: #Ecto.Schema.Metadata<:loaded, "movies">,
+  actors: [],
+  characters: [],
+  distributor: nil,
+  id: 1,
+  tagline: "Something about video games",
+  title: "Ready Player One"
+}
+```
+
+Next up, we'll create a changeset our of our movie record:
+
+```elixir
+iex> movie_changeset = Ecto.Changeset.change(movie)                                                    
+#Ecto.Changeset<action: nil, changes: %{}, errors: [], data: #Moviedb.Movie<>,
+ valid?: true>
+```
+
+Now we'll pass our changeset as the first argument to `Ecto.Changeset.put_assoc/3`:
+
+```elixir
+iex> movie_actors_changeset = cs = movie_changeset |> Ecto.Changeset.put_assoc(:actors, [actor])
+#Ecto.Changeset<
+  action: nil,
+  changes: %{
+    actors: [
+      #Ecto.Changeset<action: :update, changes: %{}, errors: [],
+       data: #Moviedb.Actor<>, valid?: true>
+    ]
+  },
+  errors: [],
+  data: #Moviedb.Movie<>,
+  valid?: true
+>
+```
+
+Lastly, we'll update the given movie and actor records with this changeset:
+
+```elixir
+iex> Repo.update!(movie_actors_changeset)
+%Moviedb.Movie{
+  __meta__: #Ecto.Schema.Metadata<:loaded, "movies">,
+  actors: [
+    %Moviedb.Actor{
+      __meta__: #Ecto.Schema.Metadata<:loaded, "actors">,
+      id: 1,
+      movies: #Ecto.Association.NotLoaded<association :movies is not loaded>,
+      name: "Bob"
+    }
+  ],
+  characters: [],
+  distributor: nil,
+  id: 1,
+  tagline: "Something about video games",
+  title: "Ready Player One"
+}
+```
+
+We can see that this gives us a movie record with the new actor properly associated and already preloaded for us under `movie.actors`
+
+We can use this same approach to create a brand new actor that is associated with the given movie. Instead of passing a _saved_ actor struct into `put_assoc`, we simply pass in an actor struct describing a new actor that we want to create:
+
+```elixir
+iex> changeset = movie_changeset |> Ecto.Changeset.put_assoc(:actors, [%{name: "Gary"}])                      
+#Ecto.Changeset<
+  action: nil,
+  changes: %{
+    actors: [
+      #Ecto.Changeset<
+        action: :insert,
+        changes: %{name: "Gary"},
+        errors: [],
+        data: #Moviedb.Actor<>,
+        valid?: true
+      >
+    ]
+  },
+  errors: [],
+  data: #Moviedb.Movie<>,
+  valid?: true
+>
+iex>  Repo.update!(changeset)
+%Moviedb.Movie{
+  __meta__: #Ecto.Schema.Metadata<:loaded, "movies">,
+  actors: [
+    %Moviedb.Actor{
+      __meta__: #Ecto.Schema.Metadata<:loaded, "actors">,
+      id: 2,
+      movies: #Ecto.Association.NotLoaded<association :movies is not loaded>,
+      name: "Gary"
+    }
+  ],
+  characters: [],
+  distributor: nil,
+  id: 1,
+  tagline: "Something about video games",
+  title: "Ready Player One"
+}
+```
+
+We can see that a new actor was created with an ID of "2" and the attributes we assigned it.
+
+## Querying For Associated data
+### Preloading
```

Comment:
How in depth is it appropriate do to on the topic of querying? Maybe it should include some basic queries: "give me all the movies featuring X actor" etc. Or is that content a better fit for the querying lesson?

Reply:
I think sticking to the simplest basic examples is good for queries here. Query the data such that we get a real result, but we can save the complexities of composing a nontrivial query for other lessons.